File: 2019/day1/fuel_test.py
import fuel
import unittest


class FuelTestCase(unittest.TestCase):
    def test_simple(self):
        self.assertEqual(2, fuel.get_fuel_requirement(12))
        self.assertEqual(2, fuel.get_fuel_requirement(14))
        self.assertEqual(654, fuel.get_fuel_requirement(1969))
        self.assertEqual(33583, fuel.get_fuel_requirement(100756))


if __name__ == '__main__':
    unittest.main()

File: 2019/day1/fuel_test.dart
import "package:test/test.dart";

import 'fuel.dart';

void main() {
  test("Input simple", _testSimple);
  test("Input list", _testList);
  test("Inclusive", _testInclusive);
}

void _testSimple() {
  expect(2, getFuelRequirement(12));
  expect(2, getFuelRequirement(14));
  expect(654, getFuelRequirement(1969));
}

void _testList() {
  expect(4, getFuelRequirements([12, 14]));
}

void _testInclusive() {
  expect(2, getFuelRequirementInclusive(12));
  expect(2, getFuelRequirementInclusive(14));
  expect(966, getFuelRequirementInclusive(1969));
}
